warehouse/query-core/src/main/java/datawave/query/jexl/visitors/TreeFlatteningRebuildingVisitor.java

```diff
@@ -52,246 +80,276 @@ public TreeFlatteningRebuildingVisitor(boolean removeReferences) {
      */
     @SuppressWarnings("unchecked")
     public static <T extends JexlNode> T flattenAll(T node) {
-        return flatten(node, true);
+        return TreeFlatteningRebuilder.flattenAll(node);
     }
     
-    /**
-     * This will flatten ands and ors. If requested this will also remove reference expressions and references where possible. NOTE: If you remove reference
-     * expressions and references, this may adversely affect the evaluation of the query (true in the index query logic case: bug?).
-     */
-    @SuppressWarnings("unchecked")
-    private static <T extends JexlNode> T flatten(T node, boolean removeReferences) {
-        TreeFlatteningRebuildingVisitor visitor = new TreeFlatteningRebuildingVisitor(removeReferences);
-        
-        return (T) node.jjtAccept(visitor, null);
+    @Override
+    public ASTJexlScript apply(ASTJexlScript input) {
+        return treeFlatteningRebuilder.flattenTree(input);
     }
     
     @Override
-    public Object visit(ASTOrNode node, Object data) {
-        ASTOrNode orNode = JexlNodes.newInstanceOfType(node);
-        orNode.image = node.image;
-        
-        return flattenTree(node, orNode, data);
+    public Object visit(ASTJexlScript node, Object data) {
+        return treeFlatteningRebuilder.flattenTree(node);
```

Comment:
Won't this take care of the entire tree?  Why do we need all of the methods below.  In fact to we actually need this one?

Reply:
That was my initial thought.  Then I ran a build + tests and found that some classes are calling the visitor like so:

TreeFlatteningRebuildingVisitor visitor = new TreeFlatteningRebuildingVisitor(true);
someNode.jjtAccept(visitor, null);

There were a fair amount of classes calling the visitor this way.